src/shared_memory_manager.cc

```diff
@@ -243,32 +253,42 @@ SharedMemoryManager::RegisterSystemSharedMemory(
 
   // register
   void* mapped_addr;
-  int shm_fd = -1;
+  ShmFile* shm_file = nullptr;
```

Comment:
Guan was right :)

Reply:
nit: wrap in smart pointer and move assign to SharedMemoryInfo after successfully initialized. In this way we can move clean-up logic into its destructor to handle early return due to unexpected error.

Reply:
I ended up wrapping ShmFile in a shared pointer because a unique pointer seemed incompatible with what we were trying to achieve with the lookup. This may change depending on how we move forward and fix the current issue regarding key reuse.